api/src/middleware/check-ip.ts

```diff
@@ -7,11 +7,11 @@ export const checkIP: RequestHandler = asyncHandler(async (req, res, next) => {
 	const role = await database
 		.select('ip_access')
 		.from('directus_roles')
-		.where({ id: req.accountability!.role })
```

Comment:
We know that the accountability object exists in this part of the request flow, so we should make it explicit that it exists

Reply:
IMO, using the non-null assertion is generally bad... See https://palantir.github.io/tslint/rules/no-non-null-assertion/
But it's up to you - should I omit those changes in the new pull request?

Reply:
> Using non-null assertion cancels the benefits of the strict null checking mode.

It depends IMO. In this case, the order of when the middleware is called changes the type available on the `req` object. How would you type that otherwise?

`req.accountability.role` isn't allowed, as TS doesn't know that `req.accountability` exists at this point in the request flow, but `req.accountability?.role` inside the where query indicates that we expect it to be undefined or a value, which isn't the case either.

I guess the next best thing is to add a 

```js
if (!req.accountability) {
  throw new Error('Accountability is missing');
}
```

to please TS, and to make sure the middleware fails if it's used outside of it's intended context

Reply:
> It depends IMO. In this case, the order of when the middleware is called changes the type available on the req object. How would you type that otherwise?
>
> req.accountability.role isn't allowed, as TS doesn't know that req.accountability exists at this point in the request flow, but req.accountability?.role inside the where query indicates that we expect it to be undefined or a value, which isn't the case either.

I agree, it always depends! And it's kind of annoying...


> I guess the next best thing is to add a
> ```
> if (!req.accountability) {
>  throw new Error('Accountability is missing');
> }
> ```
> to please TS, and to make sure the middleware fails if it's used outside of it's intended context

Yes, I think this would be the best thing in this case!